addon/models/osf-model.js

```diff
@@ -33,24 +33,35 @@ export default DS.Model.extend({
     ready() {
         this._super(...arguments);
         this.set('_dirtyRelationships', Ember.Object.create({}));
-
-        this.eachRelationship((relationship, meta) => {
-            let rel = relationship;
-            this.get(rel).then(() => {
-                var watch = rel;
-                if (meta.kind === 'hasMany') {
-                    watch = `${rel}.[]`;
+    },
+    save() {
+        this.eachRelationship((rel, meta) => {
+            if (meta.kind === 'hasMany') {
+                let relation = this.hasMany(rel).hasManyRelationship;
+                if (relation.record.isLoaded()) {
+                    if (
+                        (relation.canonicalState.filter(record => record && (Object.keys(record.changedAttributes()).length > 0 || record.isNew())).length > 0) ||
+                        relation.canonicalMembers.size !== relation.members.size
+                    ) {
+                        let key = `_dirtyRelationships.${rel}`;
+                        this.set(key, true);
+                    }
                 }
-                this.addObserver(rel, () => {
-                    var key = `_dirtyRelationships.${rel}`;
-                    this.set(key, !Ember.isEmpty(this.get(key)));
-                    // Checks is_embedded flag set in osf-serializer.
-                    if (this.get('relationshipLinks') && this.get('relationshipLinks')[rel] && this.get('relationshipLinks')[rel].is_embedded) {
-                        this.set(key, false);
-                        this.get('relationshipLinks')[rel].is_embedded = false;
+            } else if (meta.kind === 'belongsTo') {
+                let relation = this.belongsTo(rel).belongsToRelationship;
+                if (relation.record.isLoaded()) {
+                    var record = relation.members.list[0];
```

Comment:
Yep (for that one it actually doesnt matter, but canonicalState makes sense). So there, you are making two different checks:
- Have the current members (canonical) been modified?
- Have we removed or added members (comparison between members and canonical).
For the belongsTo, we want to make sure that it grabs something, so members making sense (as if that member had just been added, it would not be in canonical)